src/api/spec/controllers/staging/excluded_requests_controller_spec.rb

```diff
@@ -0,0 +1,134 @@
+require 'rails_helper'
+
+RSpec.describe Staging::ExcludedRequestsController, type: :controller, vcr: true do
+  let(:user) { create(:confirmed_user, login: 'moises') }
+  let(:other_user) { create(:confirmed_user, login: 'david') }
+  let(:project) { user.home_project }
+  let(:staging_workflow) { create(:staging_workflow_with_staging_projects, project: project) }
+  let(:group) { staging_workflow.managers_group }
+  let(:manager) { create(:confirmed_user, login: 'ana', groups: [group]) }
+  let(:source_project) { create(:project, name: 'source_project') }
+  let(:target_package) { create(:package, name: 'target_package', project: project) }
+  let(:source_package) { create(:package, name: 'source_package', project: source_project) }
+  let(:review) { create(:review, by_group: group.title) }
+  let(:bs_request) do
+    create(:bs_request_with_submit_action,
+           creator: other_user,
+           target_project: project.name,
+           target_package: target_package.name,
+           source_project: source_project.name,
+           source_package: source_package.name,
+           reviews: [review])
+  end
+
+  before { login(manager) }
+
+  describe 'POST #create' do
+    context 'succeeds' do
+      subject { post :create, params: { number: bs_request.number, project_name: project, description: "I don't want to see you any more" }, format: :xml }
+
+      context 'response' do
+        before { subject }
+
+        it { expect(response).to have_http_status(:success) }
+      end
+
+      it { expect { subject }.to(change { staging_workflow.request_exclusions.count }.by(1)) }
+    end
+
+    context 'fails: project does not exist' do
+      subject { post :create, params: { number: bs_request.number, project_name: 'i_do_not_exist', description: "I don't want to see you any more" }, format: :xml }
+
+      context 'response' do
+        before { subject }
+
+        it { expect(response).to have_http_status(404) }
+      end
+
+      it { expect { subject }.not_to(change { staging_workflow.request_exclusions.count }) }
+    end
+
+    context 'fails: project without staging_workflow' do
+      let(:project_without_staging) { create(:project, name: 'no_staging') }
+      subject { post :create, params: { number: bs_request.number, project_name: project_without_staging, description: "I don't want to see you any more" }, format: :xml }
+
+      context 'response' do
+        before { subject }
+
+        it { expect(response).to have_http_status(400) }
+      end
+
+      it { expect { subject }.not_to(change { staging_workflow.request_exclusions.count }) }
+    end
+
+    context 'fails: invalid request exclusion' do
+      let(:project_without_staging) { user.home_project }
```

Comment:
Huh? https://github.com/openSUSE/open-build-service/pull/6336/files#diff-d6b6bb094c9652eb21cfdfa20e28ed77R7 here it has a staging workflow attached? How does this work?

Reply:
it seems I didn't push the most recent changes :confounded: